### ISSUE TITLE
Unhandled exception in libraries store when closing document

### ISSUE THREAD

**Issue by iwehrman** [adobe-photoshop/spaces-design#2565]
1. Open a new artboard document.
2. Close the document:

```
Executing action documents.close after waiting 1ms; 1/0
fluxcontroller.js:563 Action documents.close failed: TypeError: Cannot read property 'get' of null
    at Fluxxor.createStore._handleClosedGraphicDocument (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/src/js/stores/library.js:344:58)
    at Store.__handleAction__ (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/bower_components/fluxxor/build/fluxxor.js:618:16)
    at null.<anonymous> (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/bower_components/fluxxor/build/fluxxor.js:167:41)
    at file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/bower_components/fluxxor/build/fluxxor.js:2057:20
    at file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/bower_components/fluxxor/build/fluxxor.js:3698:12
    at baseForOwn (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/bower_components/fluxxor/build/fluxxor.js:2175:11)
    at file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/bower_components/fluxxor/build/fluxxor.js:2199:13
    at Dispatcher.doDispatchLoop (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/bower_components/fluxxor/build/fluxxor.js:150:4)
    at Dispatcher._dispatch (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/bower_components/fluxxor/build/fluxxor.js:139:11)
    at defaultDispatchInterceptor [as dispatchInterceptor] (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/bower_components/fluxxor/build/fluxxor.js:93:4)
    at Dispatcher.dispatch (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/bower_components/fluxxor/build/fluxxor.js:117:9)
    at Object.dispatchBinder.dispatch (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/bower_components/fluxxor/build/fluxxor.js:287:21)
    at Object.<anonymous> (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/src/js/actions/documents.js:412:22)
From previous event:
    at Object.disposeDocument (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/src/js/actions/documents.js:406:14)
    at FluxController._applyAction (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/src/js/fluxcontroller.js:453:36)
    at Object.create.transfer.value (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/src/js/fluxcontroller.js:387:33)
    at Object.<anonymous> (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/src/js/actions/documents.js:616:29)
From previous event:
    at Object.close (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/src/js/actions/documents.js:613:14)
    at FluxController._applyAction (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/src/js/fluxcontroller.js:453:36)
    at null.<anonymous> (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/src/js/fluxcontroller.js:546:37)
From previous event:
    at null.<anonymous> (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/src/js/fluxcontroller.js:542:22)
    at AsyncDependencyQueue._processNext (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/src/js/util/async-dependency-queue.js:291:27)
    at AsyncDependencyQueue.push (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/src/js/util/async-dependency-queue.js:143:14)
    at Object.<anonymous> (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/src/js/fluxcontroller.js:525:42)
    at Object.throttledFn (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/src/js/util/synchronization.js:66:31)
    at Object._playMenuCommand (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/src/js/actions/menu.js:236:9)
    at Object.<anonymous> (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/src/js/actions/menu.js:338:30)
    at emitEvent (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/bower_components/eventEmitter/EventEmitter.js:380:50)
    at emit (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/bower_components/eventEmitter/EventEmitter.js:407:21)
    at UI._menuEventHandler (file:///Users/wehrman/Desktop/photoshop-Release_x86_64%202/Plug-Ins/Spaces/www/bower_components/spaces-adapter/src/ps/ui.js:90:14)
```

**Comment by iwehrman** [comment]
It looks like the `LIBRARIES_UNLOADED` event is happening before the `DOCUMENT_CLOSE` event. A simple fix would just be to move the initialization of `_editStatusByDocumentID` to `_handleReset`.

**Comment by chadrolfs** [comment]
related to or dupe of #2548 #2559

**Comment by iwehrman** [comment]
I don't _think_ it's a dupe, but I could be wrong.

**Comment by iwehrman** [comment]
Fixed by #2576.
